### DISCUSSION TITLE
Activate Permissions For [System]

### DISCUSSION THREAD

**Discussion by NeuralClone** [sulu/sulu#6133] [Q&A] [answered]
I've been using Sulu 2.3.1 and trying to figure out how to activate the Sulu security system to manage permissions for pages in a webspace. Documentation on this feature seems to be rather sparse. I've already looked at [Setup a Webspace: Security (Optional)](https://docs.sulu.io/en/latest/book/webspaces.html), [Securing Your Application](https://docs.sulu.io/en/latest/cookbook/securing-your-application.html), and [User Context Caching](https://docs.sulu.io/en/latest/cookbook/user-context-caching.html).

Whenever I try to click on the Permissions tab for a particular page, Sulu spits out a not particularly helpful error: "Activate permissions for Sulu."

![permissions](https://user-images.githubusercontent.com/678221/123407831-66104200-d57a-11eb-82fb-c50cfc4a23f2.png)

Where and how can I do this? I've checked the User Roles section and I've given myself all permissions for the webspace and everything else.

![permissions_2](https://user-images.githubusercontent.com/678221/123407850-69a3c900-d57a-11eb-85f5-26334a0994e3.png)

Is there a way to force `permission-check` to be true in a webspace's configuration without using a custom security system? Or is there another setting somewhere to achieve that? All I need is the default system but it's unclear how to enable permissions with that setup. When setting `permission-check` to true, a custom system name is expected, and Sulu doesn't work as the system name.

Anyway, if anyone can help shed some light on what's going on or where I might be able to look, I'd greatly appreciate it. I'm willing to share more of my configuration and additional information if needed.

Thanks in advance!

**Comment by alexander-schranz** [comment, marked as answer]
> Sulu spits out a not particularly helpful error: "Activate permissions for Sulu."

This is no error message this is a headline you should see beneath then your sulu rules.

The `permission-check="true"` and you need to use an own security system (you can not reuse `Sulu` here as it is reserved for the sulu backend!). Example use `Website` in your webspace:

```xml
<security permission-check="true">
    <system>Website</system>
</security>
```

Also make sure as documented to run the `bin/console sulu:security:init`  command to create the anonymous roles:

After that you should see it the following way:

![permissions](https://user-images.githubusercontent.com/1698337/123413672-57924c80-d5b3-11eb-8a36-3a45d6d295b3.gif)

The `Sulu` permissions are for the sulu backend who can view, edit, delete this page, ...
The `Website` permissions are for your webspace there only view permissions exist by default.

**Reply by NeuralClone** [reply]
I suspected I was missing something really obvious. Attempting to use Sulu as my security system outside of the backend was part of it. The bigger and more embarrassing thing I missed was the activation toggle that I somehow didn't actually see to the right of the "activate permissions" text. I guess it's been a longer week than I thought...

Thank you!

**Reply by alexander-schranz** [reply]
@NeuralClone No Problem, You are welcome!

Have a nice weekend and take a rest from this week!
And if you have any issues next week let us know.
